io.openems.edge.batteryinverter.deye/src/io/openems/edge/deye/common/DeyeSunHybridImpl.java

```diff
@@ -0,0 +1,505 @@
+						m(DeyeSunHybrid.ChannelId.MICROINVERTER_EXPORT_TO_GRID_CUTOFF, new UnsignedWordElement(178)),
+						new DummyRegisterElement(179, 180),
+						m(DeyeSunHybrid.ChannelId.SOLAR_ARC_FAULT_ON, new SignedWordElement(181))),
+
+				new FC16WriteRegistersTask(189,
+						m(DeyeSunHybrid.ChannelId.GEN_CONNECTED_TO_GRID_INPUT, new UnsignedWordElement(189))
+				// m(DeyeSunHybrid.ChannelId.GEN_PEAK_SHAVING_POWER, new
+				// UnsignedWordElement(190)),
+				// m(DeyeSunHybrid.ChannelId.GRID_PEAK_SHAVING_POWER, new
+				// UnsignedWordElement(191))
+				),
+
+				new FC16WriteRegistersTask(190,
+						m(DeyeSunHybrid.ChannelId.SET_GEN_PEAK_SHAVING_POWER, new SignedWordElement(190)),
+						m(DeyeSunHybrid.ChannelId.SET_GRID_PEAK_SHAVING_POWER, new SignedWordElement(191))),
+
+				new FC16WriteRegistersTask(209,
+						m(DeyeSunHybrid.ChannelId.UPS_BACKUP_DELAY_TIME, new UnsignedWordElement(209))),
+
+				// new FC3ReadRegistersTask(214, Priority.LOW,
+				// m(SymmetricEss.ChannelId.SOC, new UnsignedWordElement(214))),
+
+				new FC16WriteRegistersTask(214, m(SymmetricEss.ChannelId.SOC, new SignedWordElement(214))),
+
+				new FC16WriteRegistersTask(340,
+						m(DeyeSunHybrid.ChannelId.MAX_SOLAR_SELL_POWER, new UnsignedWordElement(340), SCALE_FACTOR_1)),
+
+				new FC16WriteRegistersTask(347, m(DeyeSunHybrid.ChannelId.CT_RATIO, new UnsignedWordElement(347))),
+
+				new FC3ReadRegistersTask(500, Priority.LOW,
+						m(DeyeSunHybrid.ChannelId.INVERTER_RUN_STATE, new UnsignedWordElement(500))),
+
+				new FC3ReadRegistersTask(588, Priority.LOW,
+						m(DeyeSunHybrid.ChannelId.BAT_1_SOC, new UnsignedWordElement(588))),
+
+				new FC3ReadRegistersTask(607, Priority.LOW,
+						m(SymmetricEss.ChannelId.ACTIVE_POWER, new SignedWordElement(607)),
+						m(SymmetricEss.ChannelId.REACTIVE_POWER, new SignedWordElement(608))),
+
+				new FC3ReadRegistersTask(620, Priority.LOW, //
+						m(DeyeSunHybrid.ChannelId.APPARENT_POWER, new UnsignedWordElement(620)))
+
+		);
+	}
+
+	@Override
+	public String debugLog() {
+		return "SoC:" + this.getSoc().asString() //
+				+ "|L:" + this.getActivePower().asString() //
+				+ "|Active Power:" + this.channel(SymmetricEss.ChannelId.ACTIVE_POWER).value().asStringWithoutUnit()
+				+ ";" + "|Allowed:"
+				+ this.channel(ManagedSymmetricEss.ChannelId.ALLOWED_CHARGE_POWER).value().asStringWithoutUnit() + ";"
+				+ this.channel(ManagedSymmetricEss.ChannelId.ALLOWED_DISCHARGE_POWER).value().asString();
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		if (!this.isEnabled()) {
+			return;
+		}
+		switch (event.getTopic()) {
+		case EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE:
+			this.applyPowerLimitOnPowerDecreaseCausedByOvertemperatureError();
+			this.calculateEnergy();
+			break;
+		case EdgeEventConstants.TOPIC_CYCLE_BEFORE_CONTROLLERS:
+			this.defineWorkState();
+			break;
+		}
+	}
+
+	private LocalDateTime lastDefineWorkState = null;
+
+	private void defineWorkState() {
```

Comment:
Is there a "Work State" on Deye ?

Reply:
Deye does have a run state.
<img width="924" alt="Bildschirmfoto 2024-04-15 um 10 07 49" src="https://github.com/OpenEMS/openems/assets/18745016/d018c2de-b9aa-4f89-b2ce-9ce3d175f792">